test/e2e/config/e2e_conf.yaml

```diff
@@ -21,19 +21,19 @@ providers:
   - name: ${CAPI_FROM_RELEASE}
     value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/${CAPI_FROM_RELEASE}/core-components.yaml"
     type: "url"
-    contract: v1alpha4
+    contract: ${CONTRACT}
     replacements:
       - old: --metrics-addr=127.0.0.1:8080
         new: --metrics-addr=:8080
     files:
-      - sourcePath: "../data/shared/v1alpha4/metadata.yaml"
+      - sourcePath: "../data/shared/${CAPI_FROM_RELEASE:0:4}/metadata.yaml"
   - name: ${CAPI_TO_RELEASE}
     # Use manifest from source files
     value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/${CAPI_TO_RELEASE}/core-components.yaml"
     type: "url"
-    contract: v1beta1
+    contract: ${CONTRACT}
```

Comment:
I'm not understanding how this can work. How can we have the same variable for these values when they were different before?

Reply:
https://github.com/metal3-io/cluster-api-provider-metal3/pull/904/files#diff-fe6e748ccfdc1ed1783825a0c4e64208466f536d94caedb88ae0930be0581b98R64-R72

if CAPI_FROM_RELEASE is in v0.4* the var will be `v1alpha4` else `v1beta1`

Reply:
Emm when FROM_RELEASE and TO_RELEASE are different might be conflicting ! I will check it